include/NDDoubleBufferAccessorDecorator.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "NDRegisterAccessor.h"
+#include "NDRegisterAccessorDecorator.h"
+#include "NumericAddressedBackend.h"
+#include "RegisterInfo.h"
+#include "TransferElement.h"
+
+#include <string>
+
+namespace ChimeraTK {
+
+  template<typename UserType>
+  class NumericDoubleBufferAccessorDecorator : public NDRegisterAccessorDecorator<UserType> {
+   public:
+    using ChimeraTK::NDRegisterAccessorDecorator<UserType>::buffer_2D;
+    using ChimeraTK::NDRegisterAccessorDecorator<UserType>::_target;
+    NumericDoubleBufferAccessorDecorator(const boost::shared_ptr<ChimeraTK::NDRegisterAccessor<UserType>>& target,
+        std::optional<NumericAddressedRegisterInfo::DoubleBufferInfo> doubleBufferConfig,
+        const boost::shared_ptr<DeviceBackend>& backend,
+        std::shared_ptr<NumericAddressedBackend::DoubleBufferControlState> controlState);
+
+    void doPreRead(TransferType type) override;
+
+    void doReadTransferSynchronously() override;
+
+    void doPostRead(TransferType type, bool hasNewData) override;
+
+    [[nodiscard]] bool isWriteable() const override { return false; }
+
+    void doPreWrite(TransferType, VersionNumber) override {
+      throw ChimeraTK::logic_error("NumericAddressBackend DoubleBufferPlugin: Writing is not allowed atm.");
+    }
+
+    void doPostWrite(TransferType, VersionNumber) override {
+      // do not throw here again
+    }
+    // below functions are needed for TransferGroup to work
+    std::vector<boost::shared_ptr<TransferElement>> getHardwareAccessingElements() override;
+
+    std::list<boost::shared_ptr<TransferElement>> getInternalElements() override { return {}; }
+
+    void replaceTransferElement(boost::shared_ptr<ChimeraTK::TransferElement> /* newElement */) override {
+      // do nothing, we do not support merging of DoubleBufferAccessorDecorators
```

Comment:
We will have to support merging eventually because we don't want to do individual transfers of neighbouring registers. Once we can describe individual channels of multiplexed registers, we *have to support it*. We cannot transfer the whole DAQ area 16 times and only extract one channel.

But to be honest, I have no idea yet how to combine that with double buffering. Elements are merged when you put them into the transfer group, but we cannot put the registers of the promary and the secondary accessor into the same transfer group. So we probably need two "under the hood" transfer groups for each DoubleBufferControlState.

Now that I think about it, I don't even know if it works cleanly in the current implementation.